packages/game-passes/contracts/GamePasses.sol

```diff
@@ -619,6 +625,13 @@ contract GamePasses is
             revert ArrayLengthMismatch();
         }
 
+        // Check for same token IDs in burn and mint operations
+        for (uint256 i; i < burnTokenIds.length; i++) {
+            if (burnTokenIds[i] == mintTokenIds[i]) {
+                revert SameTokenIdBurnAndMint(burnTokenIds[i]);
+            }
+        }
+
```

Comment:
Iḿ not sure that there is a problem when `burnId == mintId`, `checkMaxSupply` is not affected because you don't discount on burn, is just counting how much you mint. If there is an edge case what happen when `burnTokenIds[ someIdx ] == burntokenIds[ anotherIdx ]` is not the same case ?
Maybe just adding a `@dev` comment is enough.

Reply:
Hmm, I was thinking that it may prevent accidental operations, but I do agree its not a big issue and yes it will still work if the ids have different positions in the array.

Im on the fence whether this should be kept or removed, I see pros and cons.
@atkinsonholly, what do you think?

Reply:
Actually we can do that check on the backend, I will remove it.